### ISSUE TITLE
Add `unsafe` snippet

### ISSUE THREAD

**Issue by pvdrz** [rust-analyzer/rust-analyzer#10392]
Hi!

I'd like to propose (and in case is considered reasonable, also I'd like to implement it myself) to add a snippet to surround expressions automatically in an unsafe block. Something that when autocompleting `$expr.unsafe` becomes `unsafe { $expr }`. 

This is something that I think it could be useful when dealing with a lot of intrinsics or in general any low-level API with a lot of `unsafe` function.

**Comment by lnicola** [comment]
We already have a "missing unsafe" diagnostic, and proposing a fix would probably work here. Postfix completions aren't great, since they pollute the completion list and/or we'd need to do the same work as the diagnostic to see if it should be applicable.

**Comment by pvdrz** [comment]
Oh yeah the fix sounds nice if the postfix completions aren't a good idea.

**Comment by matklad** [comment]
Yeah, I don't think unsafe is frequent enough to warrant postfix completion, and the fixit is the right UI to solve the underlying problem here.
